ci/test_spack.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
```

Comment:
This feels like a lot of bash code to achieve something that could be done more concisely in the workflow file :thinking:

Reply:
Having this as a script file allows for local testing as well

Reply:
Yeah, I think that's a strong enough argument.